mailserver/mailserver.go

```diff
@@ -357,13 +359,29 @@ func (s *WMailServer) sendHistoricMessageResponse(peer *whisper.Peer, request *w
 	return s.w.SendHistoricMessageResponse(peer, payload)
 }
 
+// openEnvelope tries to decrypt an envelope, first based on asymetric key (if
+// provided) and second on the symetric key (if provided)
+func (s *WMailServer) openEnvelope(request *whisper.Envelope) *whisper.ReceivedMessage {
```

Comment:
btw there are request.OpenSymmetric and OpenAssymetric. just in case if we dont need filters for anything except decrypting envelopes

Reply:
Yep, would look simpler storing the keys instead of filters. However, `OpenSymetric` and `OpenAsymetric` are only trying to decrypt the envelope with a particular key, on the other hand `Open` method is also [populating the received message](https://github.com/ethereum/go-ethereum/blob/master/whisper/whisperv6/envelope.go#L232) with extra fields required on the mail server in case of success.

Reply:
As long as we use `Envelope` data type, I would keep using `Open` as it does a few additional things and it might be confusing when used later by the caller (the same data type but once with fields being set and once without them).